src/main/java/de/zalando/aruha/nakadi/controller/EventStreamController.java

```diff
@@ -0,0 +1,156 @@
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.StreamingResponseBody;
+import org.zalando.problem.Problem;
+
+import javax.annotation.Nullable;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static javax.ws.rs.core.Response.Status.BAD_REQUEST;
+import static javax.ws.rs.core.Response.Status.INTERNAL_SERVER_ERROR;
+import static javax.ws.rs.core.Response.Status.NOT_FOUND;
+import static javax.ws.rs.core.Response.Status.PRECONDITION_FAILED;
+import static javax.ws.rs.core.Response.Status.SERVICE_UNAVAILABLE;
+import static org.zalando.problem.MoreStatus.UNPROCESSABLE_ENTITY;
+
+@RestController
+public class EventStreamController {
+
+    private final TopicRepository topicRepository;
+
+    private final ObjectMapper jsonMapper;
+
+    private final EventStreamFactory eventStreamFactory;
+
+    public EventStreamController(final TopicRepository topicRepository, final ObjectMapper jsonMapper,
+                                 final EventStreamFactory eventStreamFactory) {
+        this.topicRepository = topicRepository;
+        this.jsonMapper = jsonMapper;
+        this.eventStreamFactory = eventStreamFactory;
+    }
+
+    @Timed(name = "stream_events_for_event_type", absolute = true)
+    @RequestMapping(value = "/event-types/{name}/events", method = RequestMethod.GET)
+    public StreamingResponseBody streamEvents(
+            @PathVariable("name") final String eventTypeName,
+            @RequestParam(value = "batch_limit", required = false, defaultValue = "1") final int batchLimit,
+            @RequestParam(value = "stream_limit", required = false, defaultValue = "0") final int streamLimit,
+            @RequestParam(value = "batch_flush_timeout", required = false, defaultValue = "30") final int batchTimeout,
+            @RequestParam(value = "stream_timeout", required = false, defaultValue = "0") final int streamTimeout,
+            @RequestParam(value = "stream_keep_alive_limit", required = false, defaultValue = "0") final int streamKeepAliveLimit,
+            @Nullable @RequestHeader(name = "X-nakadi-cursors", required = false) final String cursorsStr,
+            final NativeWebRequest request, final HttpServletResponse response) throws IOException {
+
+        return outputStream -> {
+            try {
+                // todo: we should get topic from EventType after persistence of EventType is implemented
+                @SuppressWarnings("UnnecessaryLocalVariable")
+                final String topic = eventTypeName;
+
+                // validate parameters
+                if (!topicRepository.topicExists(topic)) {
```

Comment:
Yes, I don't think performance is a huge concern right now, but I was curious if is this really required. I thought that kafka would throw some specific error in case the topic wasn't found. But kafka and it's api is not that familiar for me yet.